cfme/v2v/migrations.py

```diff
@@ -0,0 +1,417 @@
+
+
+class InfraMappingWizardGeneralView(View):
+    name = TextInput(name='name')
+    description = TextInput(name='description')
+    include_buttons = View.include(InfraMappingCommonButtonsSet1)
+
+
+class InfraMappingWizardClustersView(View):
+    include_buttons_set1 = View.include(InfraMappingCommonButtonsSet1)
+    include_buttons_set2 = View.include(InfraMappingCommonButtonsSet2)
+    source_clusters = MultiSelectList('source_clusters')
+    target_clusters = MultiSelectList('target_clusters')
+
+    @property
+    def is_displayed(self):
+        return self.source_clusters.item_count > 0 or self.target_clusters.item_count > 0
+
+    def fill(self, values, partial_match=False):
+        """Use _add_cluster_mapping to add all mappings specified in values.
+
+        Args:
+        values:
+            format as following:
+                {
+                    'mapping1':
+                    {
+                        'sources':['item1','item2'],
+                        'target': 'item_target1'
+                    },
+                    'mapping2':
+                    {
+                        'sources':['item3','item4'],
+                        'target': 'item_target2'
+                    },
+                    ...
+                }
+        """
+        try:
+            for mapping in values:
+                self._add_cluster_mapping(values[mapping]['sources'], values[mapping]['target'],
+                    partial_match=partial_match)
+            return True
+        except Exception as e:
+            self.logger.debug("Following Exception occured while trying to fill:%s", e)
+            return False
+
+    def _add_cluster_mapping(self, sources=None, target=None, partial_match=False):
+        """Add mapping from one or more sources to one target.
+
+        Args:
+            sources: list of source items
+            target: str containing target item
+        """
+
+        sources = sources or []
+        if isinstance(sources, list) and sources and target:
+            for item in sources:
+                self.logger.info("Mapping Source Cluster %s to Target Cluster %s", item, target)
+                self.source_clusters.select_by_text(item, partial_match=partial_match)
+                self.target_clusters.select_by_text(target, partial_match=partial_match)
+                self.add_mapping.click()
+
+
+class InfraMappingWizardDatastoresView(View):
+    include_buttons_set1 = View.include(InfraMappingCommonButtonsSet1)
+    include_buttons_set2 = View.include(InfraMappingCommonButtonsSet2)
+    source_datastores = MultiSelectList('source_datastores')
+    target_datastores = MultiSelectList('target_datastores')
+    cluster_selector = BootstrapSelect(locator='.//div[contains(@class,"cluster_select_select")]')
+
+    @property
+    def is_displayed(self):
+        return self.source_datastores.item_count > 0 or self.target_datastores.item_count > 0
+
+    def fill(self, values, partial_match=False):
```

Comment:
**REQUIRED**
Please implement in the same as in `InfraMappingWizardClustersView`